javacomparator/DiffGenerator.java

```diff
@@ -0,0 +1,457 @@
+     *
+     * @param data     The row data for the table formed
+     * @param header   The Headers for the table
+     * @param filePath Path to the XLSX file to be stored
+     */
+    private void writeCSVFile(ArrayList<String[]> data, String[] header, String filePath) {
+        Workbook workbook = new XSSFWorkbook();
+        Sheet sheet = workbook.createSheet("Data");
+
+        Font headerFont = workbook.createFont();
+        headerFont.setBold(true);
+        headerFont.setFontHeightInPoints((short) 14);
+        headerFont.setColor(IndexedColors.RED.getIndex());
+
+        CellStyle headerCellStyle = workbook.createCellStyle();
+        headerCellStyle.setFont(headerFont);
+
+        // Create a Row
+        Row headerRow = sheet.createRow(0);
+
+        for (int i = 0; i < header.length; i++) {
+            Cell cell = headerRow.createCell(i);
+            cell.setCellValue(header[i]);
+            cell.setCellStyle(headerCellStyle);
+        }
+
+        // Create Other rows and cells with contacts data
+        int rowNum = 1;
+
+        for (String[] entry : data) {
+            Row row = sheet.createRow(rowNum++);
+            for (int i = 0; i < entry.length; i++) {
+                row.createCell(i).setCellValue(entry[i]);
+            }
+        }
+
+        // Resize all columns to fit the content size
+        for (int i = 0; i < header.length; i++) {
+            sheet.autoSizeColumn(i);
+        }
+
+        // Write the output to a file
+        try {
+            FileOutputStream fileOut = new FileOutputStream(filePath);
+            workbook.write(fileOut);
+            fileOut.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Function returning the names of all JSON folders present in the folder
+     *
+     * @param directoryPath path to load the folder list from.
+     * @return List of the folders present at the path.
+     */
+    public List<String> findFoldersInDirectory(String directoryPath) {
+        File directory = new File(directoryPath);
+
+        var directoryFileFilter = new FileFilter() {
+            public boolean accept(File file) {
+                return file.isDirectory();
+            }
+        };
+
+        File[] directoryListAsFile = directory.listFiles(directoryFileFilter);
+        assert directoryListAsFile != null;
+        List<String> foldersInDirectory = new ArrayList<>(directoryListAsFile.length);
+        for (File directoryAsFile : directoryListAsFile) {
+            foldersInDirectory.add(directoryAsFile.getName());
+        }
+
+        return foldersInDirectory;
+    }
+
+
+    /**
+     * This prepares and run the comparator on folder and files.
+     */
+    public void prepareFolderAndRun() {
+
+        Path finalFolder = Paths.get(finalOutputPath);
+        if (Files.notExists(finalFolder)) {
+            // the output path does not exist
```

Comment:
Please change all comments to start with Capital first letters and end with period (dot).